src/Components/Aspire.Azure.Messaging.EventHubs/EventProcessorClientComponent.cs

```diff
@@ -0,0 +1,69 @@
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Extensions.Hosting;
+
+internal sealed class EventProcessorClientComponent(IConfiguration builderConfiguration)
+    : EventHubsComponent<EventProcessorClient, EventProcessorClientOptions>
+{
+    // cannot be in base class as source generator chokes on generic placeholders
+    protected override void BindClientOptionsToConfiguration(IAzureClientBuilder<EventProcessorClient, EventProcessorClientOptions> clientBuilder, IConfiguration configuration)
+    {
+#pragma warning disable IDE0200 // Remove unnecessary lambda expression - needed so the ConfigBinder Source Generator works
+        clientBuilder.ConfigureOptions(options => configuration.Bind(options));
+#pragma warning restore IDE0200
+    }
+
+    protected override IAzureClientBuilder<EventProcessorClient, EventProcessorClientOptions> AddClient<TBuilder>(TBuilder azureFactoryBuilder, AzureMessagingEventHubsSettings settings,
+        string connectionName, string configurationSectionName)
+    {
+        return azureFactoryBuilder.RegisterClientFactory<EventProcessorClient, EventProcessorClientOptions>(
+            (options, cred) =>
+            {
+                var connectionString = settings.ConnectionString;
+                if (string.IsNullOrEmpty(connectionString) && string.IsNullOrEmpty(settings.Namespace))
+                {
+                    throw new InvalidOperationException(
+                        $"A EventHubProducerClient could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'ConnectionString' or 'Namespace' in the '{configurationSectionName}' configuration section.");
+                }
+
+                if (string.IsNullOrEmpty(settings.BlobClientConnectionName))
+                {
+                    // throw an invalid operation exception if the blob client connection string is not provided
+                    throw new InvalidOperationException(
+                                               $"A EventProcessorClient could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'BlobClientConnectionName' in the '{configurationSectionName}' configuration section.");
+                }
+
+                // todo: add more settings and clientoptions validation; also we should use a deterministic ID for the processor
+
+                var blobConnectionString =
+                    builderConfiguration.GetConnectionString(settings.BlobClientConnectionName);
+
+                Uri blobUri = new Uri($"{blobConnectionString}/checkpoints"); // fixme: this should be a compound name including the consumer group name ideally
+                var blobClient = new BlobContainerClient(blobUri, cred);
+                blobClient.CreateIfNotExists(); // this works, but the processor client fails to create blobs later?
```

Comment:
> blobClient.CreateIfNotExists(); // this works, but the processor client fails to create blobs later?

The processor intentionally does not create the container, so this would be needed if you're intending to create automatically.   Note that this does require permissions above the container to work.

Reply:
@jsquire Yep, I understand that the processor makes no attempt to create the container. I am actually referring to a different problem I am experiencing. If the blob container client uses a tokenidentity (and not via connectionstring) to authenticate, the processor client fails to read/create blob checkpoints with "400 Value for one of the query parameters specified in the request URI is invalid." It's very suspicious.

Reply:
^ Looking at all of the examples on learn.microsoft.com and on github, they all configure a processor with a sasuri or sharedkey connectionstring. I will try to repro independently, unless there's a known issue?

Reply:
We test nightly with token credentials as well as with connection strings.   Can't say that I've ever seen or heard of this before.   If you're able to repro and can grab the exact error message, full stack trace, and capture Azure SDK logs for that, it would help me pull the Storage team in to troubleshoot.   [Logging sample](https://learn.microsoft.com/dotnet/azure/sdk/logging)

We also have samples for identity-based connections.   For example, [this one](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample05_IdentityAndSharedAccessCredentials.md#processing-events-with-identity-based-authorization).

Reply:
![image](https://github.com/dotnet/aspire/assets/1844001/9215e47e-0803-4e4a-859c-6cce46d954ae)

This is as bare bones as it gets. Very odd.

Reply:
I switched to using the BlobUriBuilder as an example and now it works. 

```csharp
                BlobUriBuilder blobUriBuilder = new BlobUriBuilder(new Uri(blobConnectionString!));
                blobUriBuilder.BlobContainerName = "checkpoints";
                Uri blobUri = blobUriBuilder.ToUri();

                //Uri blobUri = new Uri($"{blobConnectionString}/checkpoints"); 
```

The reason is that my interpolated blobUri had two slashes before checkpoints. I did not think that would parse a valid Uri, but apparently so! But that leads to the 400 error.

Reply:
Any chance you can grab the message and stack trace after flattening the aggregate?

Reply:
> Any chance you can grab the message and stack trace after flattening the aggregate?

```
System.AggregateException
  HResult=0x80131500
  Message=One or more errors occurred. (Service request failed.
Status: 400 (Value for one of the query parameters specified in the request URI is invalid.)
ErrorCode: InvalidQueryParameterValue

Headers:
Transfer-Encoding: chunked
Server: Microsoft-HTTPAPI/2.0
x-ms-request-id: ee637926-401e-003e-3571-796f9f000000
x-ms-client-request-id: a54c06b7-723e-4236-af03-4f8ce11677db
x-ms-error-code: InvalidQueryParameterValue
Date: Mon, 18 Mar 2024 20:15:56 GMT
)
  Source=Azure.Messaging.EventHubs
  StackTrace:
   at Azure.Messaging.EventHubs.Primitives.EventProcessor`1.<StartProcessingInternalAsync>d__86.MoveNext()
   at Azure.Messaging.EventHubs.Primitives.EventProcessor`1.<StartProcessingAsync>d__61.MoveNext()
   at Azure.Messaging.EventHubs.EventProcessorClient.<StartProcessingInternalAsync>d__73.MoveNext()
   at Azure.Messaging.EventHubs.EventProcessorClient.<StartProcessingAsync>d__54.MoveNext()
   at EventHubsConsumer.Processor.<StartAsync>d__3.MoveNext() in C:\p\github\aspire\playground\AspireEventHub\EventHubsConsumer\Processor.cs:line 31
   at Microsoft.Extensions.Hosting.Internal.Host.<<StartAsync>b__15_1>d.MoveNext()
   at Microsoft.Extensions.Hosting.Internal.Host.<ForeachService>d__18`1.MoveNext()
   at Microsoft.Extensions.Hosting.Internal.Host.<StartAsync>d__15.MoveNext()
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.<RunAsync>d__4.MoveNext()
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.<RunAsync>d__4.MoveNext()
   at Program.<<Main>$>d__0.MoveNext() in C:\p\github\aspire\playground\AspireEventHub\EventHubsConsumer\Program.cs:line 42

  This exception was originally thrown at this call stack:
    [External Code]

Inner Exception 1:
RequestFailedException: Service request failed.
Status: 400 (Value for one of the query parameters specified in the request URI is invalid.)
ErrorCode: InvalidQueryParameterValue

Headers:
Transfer-Encoding: chunked
Server: Microsoft-HTTPAPI/2.0
x-ms-request-id: ee637926-401e-003e-3571-796f9f000000
x-ms-client-request-id: a54c06b7-723e-4236-af03-4f8ce11677db
x-ms-error-code: InvalidQueryParameterValue
Date: Mon, 18 Mar 2024 20:15:56 GMT
```

Reply:
I guess this is a bug because the blob container client call  `blobClient.CreateIfNotExists()` succeeds and creates a `checkpoints` container in the root despite the double slashes, but the processor chokes.

Reply:
Ooops, the inner stacktrace is:

```
   at Azure.Core.HttpPipelineExtensions.<ProcessMessageAsync>d__0.MoveNext()
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Azure.Storage.Blobs.BlobRestClient.<GetPropertiesAsync>d__14.MoveNext()
   at Azure.Storage.Blobs.Specialized.BlobBaseClient.<GetPropertiesInternal>d__124.MoveNext()
   at Azure.Storage.Blobs.Specialized.BlobBaseClient.<GetPropertiesAsync>d__123.MoveNext()
   at Azure.Messaging.EventHubs.Primitives.BlobCheckpointStoreInternal.<GetCheckpointAsync>d__20.MoveNext()
   at Azure.Messaging.EventHubs.EventProcessorClient.<GetCheckpointAsync>d__66.MoveNext()
   at Azure.Messaging.EventHubs.Primitives.EventProcessor`1.<ValidateStorageConnectionAsync>d__95.MoveNext()
   at Azure.Messaging.EventHubs.Primitives.EventProcessor`1.<ValidateProcessingPreconditions>d__72.MoveNext()
   ```